src/python/dxpy/sugar/transfers.py

```diff
@@ -0,0 +1,952 @@
+            local_filename = os.path.join(output_dir, local_filename)
+        if not os.path.exists(local_filename):
+            raise ValueError(
+                "Expected file {} does not exist after untarring file {}",
+                local_filename, remote_file.get_id()
+            )
+        return [local_filename]
+    else:
+        # Otherwise return the list of files that were unpacked from the tar file
+        with open(file_list_filename, "rt") as inp:
+            return [os.path.join(output_dir, filename.rstrip()) for filename in inp]
+
+
+def download_and_decompress_file(
+    remote_file,
+    input_filename=None,
+    local_filename=None,
+    output_dir=None,
+    project=None,
+    block=True,
+):
+    """
+    Download and decompress a gzipped file.
+
+    Args:
+        remote_file (dxpy.DXFile or dxlink): DNAnexus link or file-id of file to
+            download.
+        input_filename (str): Name to use for the input filename, if different than
+            the name of the input_file. If not provided, platform filename is used.
+        local_filename (str): Local filename/dirname. If not None, this file/directory
+            must exist after unpacking the archive or an error is raised.
+        output_dir (str): Download file to a specific directory (default is the current
+            directory).
+        project (str): The ID of the project that contains the file, if it is not the
+            currently selected project and is not specified in the remote file
+            object/link.
+        block (bool): Wait for the download to complete before returning.
+
+    Returns:
+        The path of the decompressed file, if `block is True`, otherwise
+        a :class:`dxpy.sugar.processing.Processes` object.
+    """
+    remote_file = _as_dxfile(remote_file, project)
+
+    if input_filename is None:
+        input_filename = remote_file.describe()["name"]
+
+    if input_filename.endswith(".gz"):
+        exe = "gunzip"
+        ext_len = 3
+    elif input_filename.endswith(".bz2"):
+        exe = "bunzip2"
+        ext_len = 4
+    else:
+        raise ValueError(
+            "Unsupported compression method for file {}".format(input_filename)
+        )
+
+    if local_filename is None:
+        local_filename = input_filename[:-ext_len]
+    if output_dir:
+        local_filename = os.path.join(output_dir, local_filename)
+
+    cmds = [
+        ["dx", "download", remote_file.get_id(), "-o", "-"],
+        [exe],
+    ]
+
+    LOG.info(
+        "Downloading file %s to %s using command %s",
+        remote_file.get_id(),
+        local_filename,
+        cmds,
+    )
+
+    result = proc.chain_cmds(cmds, stdout=local_filename, block=block)
+
+    if block:
+        LOG.info(
+            "Completed downloading file %s to %s", remote_file.get_id(), local_filename
+        )
+        return local_filename
+    else:
+        return result
+
+
+def _as_dxfile(fileobj, project=None):
```

Comment:
It seems to me this function duplicates dxpy.get_handler()

Reply:
Oops - missed that. Changed it to call dxpy.get_handler unless fileobj is already a DXFile